### PR TITLE
Remove bclrewriter

### PR DESCRIPTION
**Pull request by danmosemsft**
Fixes https://github.com/dotnet/coreclr/issues/9344

Fixed a few remaining diffs with the rewritten version. It is still about 40KB (1%) larger as there are still diffs for constants that are no longer inlined, and serialization helper functions.

@jkotas @stephentoub

### CODE REVIEW

File: src/mscorlib/src/System/Reflection/Assembly.cs [hunk 1]
```diff
@@ -1691,14 +1691,6 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version versi
             return retAssembly;
         }
 
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        static internal extern RuntimeAssembly nLoadImage(byte[] rawAssembly,
```
Comment (danmosemsft) [inline src/mscorlib/src/System/Reflection/Assembly.cs]: I'm not going to edit ecalllist.h right now, I think that can be cleaned up more separately.

File: src/mscorlib/src/System/Security/Permissions/HostProtectionPermission.cs [hunk 2]
```diff
@@ -41,8 +41,8 @@ internal enum HostProtectionResource
     [AttributeUsage(AttributeTargets.Method | AttributeTargets.Constructor | AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Assembly | AttributeTargets.Delegate, AllowMultiple = true, Inherited = false )] 
     [System.Runtime.InteropServices.ComVisible(true)]
     [Serializable]
-    // This needs to be in the asmmeta to enable SecAnnotate to successfully resolve and run the security rules. It gets marked
-    // as internal by BCLRewriter so we are simply marking it as FriendAccessAllowed so it stays in the asmmeta.
+    // This needs to be in the asmmeta to enable SecAnnotate to successfully resolve and run the security rules. 
+    // We are simply marking it as FriendAccessAllowed so it stays in the asmmeta.
     [System.Runtime.CompilerServices.FriendAccessAllowedAttribute]
```
Comment (jkotas) [inline src/mscorlib/src/System/Security/Permissions/HostProtectionPermission.cs]: You can delete this comment as well as FriendAccessAllowedAttribute. None of it is applicable anymore.

**Comment by jkotas** [comment]
LGTM ... once the CI is green.

File: src/mscorlib/src/System/Runtime/InteropServices/IException.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ namespace System.Runtime.InteropServices {
     [InterfaceTypeAttribute(ComInterfaceType.InterfaceIsDual)]
     [CLSCompliant(false)]
     [System.Runtime.InteropServices.ComVisible(true)]
-    public interface _Exception
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/InteropServices/IException.cs]: This whole file should be deleted - I forgot to delete it when I was cleaning up these.

**Comment by danmosemsft** [comment]
@dotnet-bot test windows_nt x64 debug build and test ("The "GeneratePackageReport" task failed unexpectedly." again.. cited separately)

**Comment by danmosemsft** [comment]
This is not going to be finished for a while (later today?) as I need to set up a Unix machine to avoid working through the linux crossgen issues one by one.

File: src/mscorlib/src/System/AppDomain.cs [hunk 4]
```diff
@@ -837,10 +837,17 @@ public Object GetData(string name)
                 }
             }
         }
-        
+       
+#if !PLATFORM_UNIX
```
Comment (jkotas) [inline src/mscorlib/src/System/AppDomain.cs]: Why do you need to do this? Can we delete it instead?

Comment (danmosemsft) [inline src/mscorlib/src/System/AppDomain.cs]: I would have to #if !UNIX several pieces of EventSource. I can do that if you prefer.

Comment (jkotas) [inline src/mscorlib/src/System/AppDomain.cs]: What was the actual problem with it on Unix? We have `GetCurrentThreadId` in the CoreCLR Unix PAL, so it should have worked just fine as is.

Comment (danmosemsft) [inline src/mscorlib/src/System/AppDomain.cs]: It appears this was a mistake. Reverted. Thank you.

**Comment by danmosemsft** [comment]
@dotnet-bot test osx x64 checked build and test (https://github.com/dotnet/coreclr/issues/9388)

**Comment by danmosemsft** [comment]
Good to merge?

**Pull request merged**